samples/monster_generated.h

```diff
@@ -255,46 +270,53 @@ inline flatbuffers::Offset<Weapon> CreateWeapon(flatbuffers::FlatBufferBuilder &
   return CreateWeapon(_fbb, name ? 0 : _fbb.CreateString(name), damage);
 }
 
-inline flatbuffers::Offset<Weapon> CreateWeapon(flatbuffers::FlatBufferBuilder &_fbb, const WeaponT *_o);
-
-inline std::unique_ptr<MonsterT> Monster::UnPack() const {
-  auto _o = new MonsterT();
-  { auto _e = pos(); if (_e) _o->pos = std::unique_ptr<Vec3>(new Vec3(*_e)); };
-  { auto _e = mana(); _o->mana = _e; };
-  { auto _e = hp(); _o->hp = _e; };
-  { auto _e = name(); if (_e) _o->name = _e->str(); };
-  { auto _e = inventory(); if (_e) { for (flatbuffers::uoffset_t _i = 0; _i < _e->size(); _i++) { _o->inventory.push_back(_e->Get(_i)); } } };
-  { auto _e = color(); _o->color = _e; };
-  { auto _e = weapons(); if (_e) { for (flatbuffers::uoffset_t _i = 0; _i < _e->size(); _i++) { _o->weapons.push_back(_e->Get(_i)->UnPack()); } } };
-  { auto _e = equipped_type(); _o->equipped.type = _e; };
-  { auto _e = equipped(); if (_e) _o->equipped.table = EquipmentUnion::UnPack(_e, equipped_type()); };
-  return std::unique_ptr<MonsterT>(_o);
+inline void MonsterT::UnPack(const Monster *_o) {
+  pos = _o->pos();
+  mana = _o->mana();
+  hp = _o->hp();
+  name.assign(_o->name()->c_str(), _o->name()->size());
+  inventory.clear();
+  if (_o->inventory()) {
+    inventory.reserve(_o->inventory()->size());
+    for (auto it = _o->inventory()->begin(); it != _o->inventory()->end(); ++it) {
+      auto val = *it;
```

Comment:
this may need to be `auto &` for when the vector contains e.g. structs, which we don't want to copy. Better yet, remove the variable `val` and stick `*it` straight in the `push_back`.